src/Swarm/Language/Typecheck.hs

```diff
@@ -280,8 +280,10 @@ inferModule s@(Syntax _ t) = case t of
   -- correct context when checking the right-hand side in particular.
   SBind mx c1 c2 -> do
     -- First, infer the left side.
-    Module cmda ctx1 <- inferModule c1
-    a <- decomposeCmdTy cmda
+    (a, ctx1) <- (`catchError` addLocToTypeErr c1) $ do
```

Comment:
Why do we need this here?  Just curious.

Reply:
Oh, I was just going crazy from bind chains - turns out they were missing location themselves and could not help to limit the error scope. I will make this simple again :+1: